base_layer/key_manager/src/mnemonic.rs

```diff
@@ -211,20 +212,24 @@ pub fn to_bytes(mnemonic_seq: &[String]) -> Result<Vec<u8>, MnemonicError> {
 
 /// Generates a vector of bytes that represent the provided mnemonic sequence of words using the specified language
 pub fn to_bytes_with_language(mnemonic_seq: &[String], language: &MnemonicLanguage) -> Result<Vec<u8>, MnemonicError> {
-    let mut bits: Vec<bool> = Vec::new();
+    let mut bytes: Vec<u8> = Vec::new();
+    let mut rest = 0;
+    let mut rest_bits: u8 = 0;
+
     for curr_word in mnemonic_seq {
-        match find_mnemonic_index_from_word(curr_word, language) {
-            Ok(index) => {
-                let curr_bits = checked_uint_to_bits(index, 11).ok_or(MnemonicError::IntToBitsConversion)?;
-                bits.extend(curr_bits.iter().cloned());
-            },
-            Err(err) => return Err(err),
+        let index = find_mnemonic_index_from_word(curr_word, language)?;
```

Comment:
This is really clever :) A few notes:

1. It's a bit hard to follow the algo, could you write a brief comment on how it works in the function doc?
2. This takes in an arbitrary length mnemonic and this function is public and could be used to encode any length. I removed the 33 length conditional at the bottom and wrote a quick fuzzer and found some failure cases. Could you include this in your PR and see if you can fix it?

```rust
     #[test]
    fn fuzzer() {
        use rand::RngCore;
        for len in 0..1024 * 1024 {
            let mut secretkey_bytes = vec![0u8; len]; // RistrettoSecretKey::random(&mut OsRng).to_vec();
            OsRng.fill_bytes(&mut secretkey_bytes);
            let mnemonic_seq = mnemonic::from_bytes(secretkey_bytes.clone(), &MnemonicLanguage::English).unwrap();
            let mnemonic_bytes = mnemonic::to_bytes(&mnemonic_seq).unwrap();
            assert_eq!(secretkey_bytes, mnemonic_bytes, "failed len = {}", len);
        }
    }
```

Reply:
I've added arbitrary length.
But I have to rewrite the test to take only multiple of 11. Because the `from_bytes` is doing padding. So the `to_bytes` produce the padded value, which is longer (padded with 0 bytes).
Other options is to pad the secretkey_bytes with zeros to match the length of the mnemonic_bytes.  But that just complicates the test.

Reply:
Thanks :) Yeah happy with this approach